course/src/main/java/org/openedx/course/presentation/unit/html/HtmlUnitFragment.kt

```diff
@@ -210,14 +211,12 @@ private fun HTMLContentView(
                         onWebPageLoaded()
 
                         evaluateJavascript(
-                            """
-                            ${'$'}(document).ajaxSuccess(function(event, request, settings)  {
-                                if (settings.url.includes("publish_completion") && 
-                                    request.responseText.includes("ok")) {
-                                    javascript:window.callback.completionSet();
-                                }
-                            });
-                        """.trimIndent(), null
+                            context.assets.readAsText("js_injection/completions.js"),
+                            null
+                        )
+                        evaluateJavascript(
+                            context.assets.readAsText("js_injection/survey_css.js"),
+                            null
```

Comment:
We should move the file read operations to ViewModel so we won't have to read the file on every orientation change or swipe-to-refresh in future.

Reply:
I've added a ViewModel for HtmlUnitFragment and plus moved all injections from the HtmlUnitFragment class to this ViewModel
done